connectivity/connectivity-demos-test/src_test/rest/bpm/TestAsyncInvocation.java

```diff
@@ -22,12 +30,17 @@ void callBatchAsync(BpmClient bpmClient) throws InterruptedException
     ExecutionResult result = bpmClient.start().process("rest/asyncInvocation/callBatchAsync.ivp").execute();
     assertThat(result).isNotNull();
     
-    Thread.sleep(4000);
-    SignalEventQuery query = Ivy.wf().signals().history()
-            .createSignalEventQuery();
-    var fired = new ArrayList<>(query.executor().results());
-    assertThat(fired).isNotEmpty();
-    assertThat(fired.get(0).getSignalCode().toString()).isEqualTo("rest:batch:result:ok");
+    SignalEventQuery query = Ivy.wf().signals().history().createSignalEventQuery();
+    
+    Supplier<List<String>> signals = () -> new ArrayList<>(query.executor().results())
+      .stream().map(ISignalEvent::getSignalCode).map(SignalCode::toString)
+      .toList();
+    Awaitility.await()
+      .atLeast(3, TimeUnit.SECONDS) // default delay: on process being called
+      .atMost(10, TimeUnit.SECONDS)
+      .until(() -> signals.get(), 
+        codes -> !codes.isEmpty() && Objects.equals(codes.get(0), "rest:batch:result:ok")
+      );
```

Comment:
now it's easy and apparent, that we are dealing with timing issues here :nerd_face: